supervisor/shared/web_workflow/web_workflow.c

```diff
@@ -852,10 +891,16 @@ static void _reply_with_diskinfo_json(socketpool_socket_obj_t *socket, _request
 
     uint16_t total_size = fs->n_fatent - 2;
 
+    const char *writable = "false";
+    if (!_usb_active()) {
+        writable = "true";
+    }
     mp_printf(&_socket_print,
-        "{\"free\": %d, "
+        "[{\"root\": \"/\", "
+        "\"free\": %d, "
         "\"block_size\": %d, "
-        "\"total\": %d}", free_clusters * block_size, block_size, total_size * block_size);
+        "\"writable\": %s, "
+        "\"total\": %d}]", free_clusters * block_size, block_size, writable, total_size * block_size);
 
     // Empty chunk signals the end of the response.
     _send_chunk(socket, "");
```

Comment:
Does this force a flush too?

Reply:
Yes, `send_chunk("")` flushes. https://github.com/adafruit/circuitpython/pull/8667/files#diff-43f36ea6fd6b4d28f257e55f3f1f29a9890e72c88d7b56164d77ab5691b704ccR443